tests/integration_tests/functional/test_snapshot_version.py

```diff
@@ -31,18 +31,48 @@ def _create_and_start_microvm_with_net_devices(test_microvm,
         )
     test_microvm.start()
 
-    ssh_connection = net_tools.SSHConnection(test_microvm.ssh_config)
-    # Verify if guest can run commands.
-    exit_code, _, _ = ssh_connection.execute_command("sync")
-    assert exit_code == 0
+    if network_config is not None:
+        ssh_connection = net_tools.SSHConnection(test_microvm.ssh_config)
+        # Verify if guest can run commands.
+        exit_code, _, _ = ssh_connection.execute_command("sync")
+        assert exit_code == 0
+
+
+def test_create_past_versions(test_microvm_with_ssh):
+    """Test scenario: restore in previous versions."""
+    test_microvm = test_microvm_with_ssh
+
+    _create_and_start_microvm_with_net_devices(test_microvm)
+
+    snapshot_builder = SnapshotBuilder(test_microvm)
+    # Create directory and files for saving snapshot state and memory.
+    _snapshot_dir = snapshot_builder.create_snapshot_dir()
+
+    # Pause microVM for snapshot.
+    response = test_microvm.vm.patch(state='Paused')
+    assert test_microvm.api_session.is_status_no_content(response.status_code)
+
+    response = test_microvm.snapshot.create(
+        mem_file_path="/snapshot/vm.vmstate",
+        snapshot_path="/snapshot/vm.mem",
+        version="0.23.0"
+    )
+    if platform.machine() == "x86_64":
+        assert test_microvm.api_session.is_status_no_content(
```

Comment:
Since this test always passes on x86_64 I suggest a diff snapshot so it runs faster.

Reply:
Ok